braid/braid_status.c

```diff
@@ -224,6 +227,72 @@ braid_StatusGetCallingFunction(braid_Status status,
    return _braid_error_flag;
 }
 
+braid_Int
+braid_StatusGetDeltaRank(braid_Status status,
+                         braid_Int   *rank_ptr
+                         )
+{
+   braid_Basis ba = _braid_StatusElt(status, lvectors);
+   if (ba != NULL)
+   {
+      *rank_ptr = ba->rank;
+   }
+   else
+   {
+      *rank_ptr = 0;
+   }
+   return _braid_error_flag;
+}
+
+braid_Int
+braid_StatusGetBasisVec(braid_Status  status,
+                        braid_Vector *v_ptr,
+                        braid_Int     index
+                        )
+{
+   braid_Basis ba = _braid_StatusElt(status, lvectors);
+   if ((ba != NULL) && (index < ba->rank))
+   {
+      *v_ptr = ba->userVecs[index];
+   }
+   else // gracefully return NULL
+   {
+      v_ptr = NULL;
+   }
+   return _braid_error_flag;
+}
+
+braid_Int
+braid_StatusGetLocalLyapExponents(braid_Status   status,
+                                  braid_Real    *exp_ptr,
+                                  braid_Int     *num_returned)
+{
+   if (_braid_StatusElt(status, wrapper_test))
+   {
+      *num_returned = 0;
+      return _braid_error_flag;
+   }
+
+   _braid_Grid **grids = _braid_StatusElt(status, grids);
+   braid_Int level   = _braid_StatusElt(status, level);
+   braid_Int nlevels = _braid_StatusElt(status, nlevels);
+   braid_Int index   = _braid_StatusElt(status, idx);
+   braid_Int cfactor = _braid_GridElt(grids[level], cfactor);
+
+   if (!_braid_StatusElt(status, lyap_exp) ||_braid_StatusElt(status, level) > 0 || (_braid_IsFPoint(index, cfactor) && nlevels > 1))
+   {
+      *num_returned = 0;
+      return _braid_error_flag;
+   }
+
+   braid_Int clower = _braid_GridElt(grids[0], clower);
+   braid_Int i_c    = (index - clower)/_braid_GridElt(grids[0], cfactor);
```

Comment:
Can you assign "i_c" using this macro from braid.h?  If so, we should probably use the macro

+++++
#define _braid_MapFineToCoarse(findex, cfactor, cindex) \
( cindex = (findex)/(cfactor) )
++++++